DatadogCore/Sources/Datadog.swift

```diff
@@ -169,6 +163,27 @@ public struct Datadog {
             self.encryption = encryption
             self.serverDateProvider = serverDateProvider ?? DatadogNTPDateProvider()
         }
+
+        // MARK: - Internal
+
+        /// Obtains OS directory where SDK creates its root folder.
+        /// All instances of the SDK use the same root folder, but each creates its own subdirectory.
+        internal var systemDirectory: () throws -> Directory = { try Directory.cache() }
+
+        /// Default process information.
+        internal var processInfo: ProcessInfo = .processInfo
+
+        /// Sets additional configuration attributes.
+        /// This can be used to tweak internal features of the SDK.
+        internal var additionalConfiguration: [String: Any] = [:]
+
+        /// Default date provider used by the SDK and all products.
+        internal var dateProvider: DateProvider = SystemDateProvider()
+
+        /// Creates `HTTPClient` with given proxy configuration attributes.
+        internal var httpClientFactory: ([AnyHashable: Any]?) -> HTTPClient = { proxyConfiguration in
+            URLSessionClient(proxyConfiguration: proxyConfiguration)
+        }
```

Comment:
Lets continue to resolve the config in `Datadog.initialize` and not in `Datadog.Configuration` itself, so it's easier to read and maintain. I would remove this in favor of `URLSessionClient(proxyConfiguration: configuration .proxyConfiguration)` in `Datadog.initialize`.

Reply:
This change was motivated by testability. Notably, being able to initialize SDK using public API with swapping the single dependency in this integration-unit:
```swift
func testGivenTwoInstancesOfSDK_whenCollectingLogs_thenEachSDKUploadsItsOwnData() throws {
   let config: Datadog.Configuration = ...
   config.httpClientFactory = { _ in HTTPClientMock(responseCode: 200) }

   Datadog.initialize(with: config, trackingConsent: .granted)
   // ...
}
```
In that sense, the `Datadog.Configuration` acts as a basic DI container for passing internal dependencies and mocking them in tests. Same way as we do for `dateProvider`, `processInfo` and few other components.

Reply:
See my [comment](https://github.com/DataDog/dd-sdk-ios/pull/1413#discussion_r1295625397) below on the same topic 👍